### PR TITLE
[New Package] UDP Input

### PR DESCRIPTION
**Pull request by P1llus**
## What does this PR do?

This package adds the possibility to configure your own custom UDP input, allowing users to add custom integrations that might not already be covered by an existing package.

Tested with 7.16 and 8.0

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] If I'm introducing a new feature, I have modified the Kibana version constraint in my package's `manifest.yml` file to point to the latest Elastic stack release (e.g. `^7.13.0`).

### CODE REVIEW

File: packages/udp/data_stream/generic/agent/stream/udp.yml.hbs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+{{#each tags as |tag i|}}
+  - {{tag}}
+{{/each}}
+{{/if}}
+{{#contains "forwarded" tags}}
+publisher_pipeline.disable_host: true
+{{/contains}}
+{{#if processors}}
+processors:
+  {{processors}}
```
Comment (andrewkroh) [inline packages/udp/data_stream/generic/agent/stream/udp.yml.hbs:30]: I thought this `{{processors}}` needed to be left aligned to function properly.

File: packages/udp/data_stream/generic/manifest.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  - input: udp
+    description: Collect raw UDP data from listening UDP port with Elastic Agent.
+    title: Custom UDP Logs
+    template_path: udp.yml.hbs
+    vars:
+      - name: host
+        type: text
+        title: Host
+        description: |
+          The hostname used for binding the UDP socket.
```
Comment (andrewkroh) [inline packages/udp/data_stream/generic/manifest.yml:13]:
```suggestion
          Bind address for the listener. Use 0.0.0.0 to listen on all interfaces.
```

Comment (andrewkroh) [reply packages/udp/data_stream/generic/manifest.yml:13]: Maybe these two should be called `listen_address` and `listen_port`? WDYT?

**Comment by P1llus** [comment]
> Was there not already decided to include the user selectable option to keep the original message on all integrations?

Sure thing @zez3 .

Since these are pure input packages, the incoming data is always the raw format, it is up to the end-user to parse/transform the data in the way that they want, and are then also free to keep the original.

**Review by andrewkroh — APPROVED** [review]

**Review by endorama — APPROVED** [review]
:+1: on title and description!

**Comment by zez3** [comment]
> > Was there not already decided to include the user selectable option to keep the original message on all integrations?
> 
> Sure thing @zez3 .
> 
> Since these are pure input packages, the incoming data is always the raw format, it is up to the end-user to parse/transform the data in the way that they want, and are then also free to keep the original.

But the commit https://github.com/elastic/integrations/pull/2170/commits/1b5dec19cd9fbec7a9ab38d1710dfa7d9e26d6d9

removes the original message? Or am I missing something here

**Comment by P1llus** [comment]
> > > Was there not already decided to include the user selectable option to keep the original message on all integrations?
> > 
> > 
> > Sure thing @zez3 .
> > Since these are pure input packages, the incoming data is always the raw format, it is up to the end-user to parse/transform the data in the way that they want, and are then also free to keep the original.
> 
> But the commit [1b5dec1](https://github.com/elastic/integrations/commit/1b5dec19cd9fbec7a9ab38d1710dfa7d9e26d6d9)
> 
> removes the original message? Or am I missing something here

Hello @zez3 . That is because when I make packages I simply copy/paste an existing package usually. The input only packages do not need this part, as it will always come in raw.
That setting is not for the input to keep the original, is for the ingest pipelines to make sure to leave a copy of the original event before it starts modifying it.

**Pull request merged**